### ISSUE TITLE
Core/Creature: Wrong rounded health value from HealthModifier when regen is disabled

### ISSUE THREAD

**Issue by CraftedRO** [TrinityCore/TrinityCore#30447]
### Description

Probably really cosmetic issue and almost no relevant but well.
Smith Argus <Blacksmithing Trainer> is a npc in Goldshire, in  db / curhealth it does have 664 health but ingame 665, now probably  1 hp is not a big deal but it's a bit strange.

### Expected behaviour

Ingame it shoud also have 664 health value as can be seen here:

![Captură de ecran 2024-11-17 193415](https://github.com/user-attachments/assets/221890a6-4f62-48be-b2a4-b94475bb78f6)

https://www.youtube.com/watch?v=o3Aht1CAqZY
https://www.wowhead.com/wotlk/npc=514/smith-argus

### Steps to reproduce the problem

.go c id 514
.npc info
notice health

### Branch

3.3.5

### TC rev. hash/commit

32b54ffb66676ec47d7153c1b897770bc5d08493

### Operating system

Windows 11 x64

### Custom changes

None

**Comment by CraftedRO** [comment]
For some reason any value we set in db/curhealth for this npc is ignored.

**Comment by Shauren** [comment]
This column is used only if health regen is disabled

**Comment by CraftedRO** [comment]
so shoud we disable that, then ?

**Comment by Shauren** [comment]
No, this is probably due to rounding (HealthModifier is 1.02 in creature_template)
